gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskInfo.java

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Disk;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine persistent disk. A disk can be used as primary storage for your virtual
+ * machine instances.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/">Block Storage</a>
+ */
+public class DiskInfo implements Serializable {
+
+  static final Function<Disk, DiskInfo> FROM_PB_FUNCTION =
+      new Function<Disk, DiskInfo>() {
+        @Override
+        public DiskInfo apply(Disk pb) {
+          return DiskInfo.fromPb(pb);
+        }
+      };
+  static final Function<DiskInfo, Disk> TO_PB_FUNCTION =
+      new Function<DiskInfo, Disk>() {
+        @Override
+        public Disk apply(DiskInfo diskType) {
+          return diskType.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -7173418340679279619L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private final String id;
```

Comment:
Ah, actually this is something you plan on fixing in the future, and not in this PR, correct?